nautobot/utilities/tests/test_forms.py

```diff
@@ -429,3 +433,54 @@ def test_invalid_input(self):
         for test in tests:
             with self.assertRaises(forms.ValidationError):
                 self.field.clean(test)
+
+
+class AddressFieldMixinTest(TestCase):
+    """Test cases for the AddressFieldMixin."""
+
+    def setUp(self):
+        """Setting up shared variables for the AddressFieldMixin."""
+        self.ip = IPAddress.objects.create(address="10.0.0.1/24")
+        self.initial = {"address": self.ip}
+
+    def test_address_initial(self):
+        """Ensure initial kwargs for address is passed in."""
+        with mock.patch("nautobot.utilities.forms.forms.forms.ModelForm.__init__") as mock_init:
+            AddressFieldMixin(initial=self.initial, instance={})
+            mock_init.assert_called_with(initial=self.initial, instance=mock.ANY)
+
+    def test_address_instance(self):
+        """Ensure override with computed field when initial kwargs for address is not passed in."""
+
+        class DummyInstance:
+            address = self.ip
+
+        # Mock the django.forms.ModelForm __init__ function used in nautobot.utilities.forms.forms
+        with mock.patch("nautobot.utilities.forms.forms.forms.ModelForm.__init__") as mock_init:
+            AddressFieldMixin(instance=DummyInstance())
+            mock_init.assert_called_with(initial=self.initial, instance=mock.ANY)
```

Comment:
Shouldn't need a `DummyInstance` class at all:

```suggestion
        # Mock the django.forms.ModelForm __init__ function used in nautobot.utilities.forms.forms
        with mock.patch("nautobot.utilities.forms.forms.forms.ModelForm.__init__") as mock_init:
            AddressFieldMixin(instance=self.ip)
            mock_init.assert_called_with(initial=self.initial, instance=self.ip)
```